src/pipeline/grafter/pipeline/types.clj

```diff
@@ -228,16 +239,23 @@
   pipeline. The metadata map must contain a key-value pair for each
   named parameter in the pipeline function argument list. The value
   corresponding to each key in the metadata map is expected to be a
-  String describing the parameter."
-  [sym type-list metadata]
+  String describing the parameter. The opts map can contain
+  an optional :supported-operations key associated to a collection
+  containing :append and/or :delete. These operations indicate whether
+  the data returned from the pipeline can be appended to or deleted
+  from the destination."
+  [sym type-list metadata opts]
   (let [def-var (resolve-var *ns* sym)
         def-meta (meta def-var)
         arg-list (first (:arglists def-meta))
         {:keys [arg-types return-type]} (parse-type-list type-list)
         pipeline-type (pipeline-type-from-return-type-sym return-type)
-        args (resolve-pipeline-arg-descriptors arg-list arg-types metadata)]
+        args (resolve-pipeline-arg-descriptors arg-list arg-types metadata)
+        supported-operations (:supported-operations opts #{:append})]
+    (validate-supported-pipeline-operations! supported-operations)
      {:var def-var
       :doc (or (:doc def-meta) "")
       :args args
       :type pipeline-type
-      :declared-args arg-list}))
+      :declared-args arg-list
+      :operations supported-operations}))
```

Comment:
Just wondering what the rationale is for using a different key `:operations` for `:supported-operations` here.  If I understand this correctly the `:operations` key will then be exposed to PMD.  I'm thinking we should use the same key throughout for consistencies sake.